### PR TITLE
1818/remove dependencies from apply

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue

Addresses #1818

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Removes backend dependencies from the Apply component. This cascaded to the Waitlist component and the ApplicationSection component (the latter didn't seem to be doing anything besides one line of formatting so I deleted it).

Essentially I broke up the Apply component into two different components because it was large and confusing and really had two distinct sections, so I now have a "GetApplication" component and a "SubmitApplication" component that are just the Apply component split into two. All the props are new so that we're not pulling them off the listing. I also removed moment so all the Dates are passed as pre-formatted strings. I also did some generic cleanup.

All the business-logicy bits that were in Apply I moved into ListingView. There is a large diff but very little code changed, just different props and moving code to different files.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [x] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

I tested every dev listing we have by screen-for-screen comparing each listing to the one on our dev environment and they all look identical. We may need some more seed data variety to encompass every toggleable section in the sidebar. I also wrote more stories and tests for the sidebar.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes
- [ ] I have run `yarn generate:client` if I made backend changes

### CODE REVIEW

File: sites/public/src/ListingView.tsx [hunk 1]
```diff
@@ -176,6 +186,119 @@ export const ListingView = (props: ListingProps) => {
...
+                  paperApp?.file?.fileId || "",
+                  process.env.cloudinaryCloudName || ""
+                ),
+            languageString: t(`languages.${paperApp.language}`),
+          }
+        }) ?? null
+    )
+  }
+
+  const applySidebar = () => (
```
Comment (emilyjablonski) [inline sites/public/src/ListingView.tsx:261]: This replaces `Application Section`, and `Applications` and `ApplicationAddresses` replace `Apply

**Review by jaredcwhite — COMMENTED** [review]
Nice @emilyjablonski! Feels really clean. I think my only comment is I think the majority of new functions added to `ListingView` would make sense in the new shared helpers package (and some I can recognize from the group I'm planning to move directly there from `ui-components`). So, depending on how vital this is to merge in, I guess we'd have to decide if we leave it as-is with big TODO comment block to leverage the shared package when it's ready…or wait and then refactor using the shared package.

**Comment by emilyjablonski** [comment]
Since we don't have an ETA on the shared package and this has been ready to merge for a while I'd vote to merge and TODO but am flexible :)

**Comment by seanmalbert** [comment]
> Since we don't have an ETA on the shared package and this has been ready to merge for a while I'd vote to merge and TODO but am flexible :)

I'm on board with adding a TODO and merging.

**Review by seanmalbert — APPROVED** [review]

**Pull request merged**